src/main/scala/mesosphere/marathon/upgrade/DeploymentActor.scala

```diff
@@ -85,7 +85,7 @@ class DeploymentActor(
           case StartApplication(app, scaleTo) => storeAndThen(app) { startApp(app, scaleTo) }
           case ScaleApplication(app, scaleTo) => storeAndThen(app) { scaleApp(app, scaleTo) }
           case RestartApplication(app)        => storeAndThen(app) { restartApp(app) }
-          case StopApplication(app)           => stopApp(app)
+          case StopApplication(app)           => storeAndThen(app.copy(instances = 0)) { stopApp(app) }
```

Comment:
This change doesn't have any automatic tests associated with it. If I revert the change, all tests still work.